src/Futura_backend/src/lib.rs

```diff
@@ -1,43 +1,106 @@
-use std::collections::HashMap;
-use candid::Principal; // Use Principal from candid crate
+use candid::{CandidType, Principal}; // Add CandidType
+use ic_cdk::storage; // Storage for the canister
+use ic_cdk_macros::{post_upgrade, pre_upgrade}; // Macros for upgrade functions
 use lazy_static::lazy_static; // For creating a static, global storage
-use std::sync::Mutex; // Mutex for thread safety in global storage
-
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::Mutex; // Mutex for thread safety in global storage // For serializing and deserializing data
 
 lazy_static! {
     static ref MEMORY_STORAGE: Mutex<HashMap<Principal, Memory>> = Mutex::new(HashMap::new());
 }
 
-
 #[ic_cdk::query]
 fn greet(name: String) -> String {
     format!("Hello, {}!", name)
 }
 
 // Define a struct for memory
-#[derive(Clone)]
+
+#[derive(Clone, Serialize, Deserialize, CandidType)]
 struct Memory {
-	memory: String, 
+    texts: Option<Vec<Text>>,   // Change to Option to allow for optionality
+    images: Option<Vec<Image>>, // Change to Option to allow for optionality
+}
+
+#[derive(Clone, Serialize, Deserialize, CandidType)]
+struct Text {
+    content: String,
+    metadata: Option<Metadata>,
+}
+
+#[derive(Clone, Serialize, Deserialize, CandidType)]
+struct Image {
+    content: Vec<u8>,
+    metadata: Option<Metadata>,
 }
 
+#[derive(Clone, Serialize, Deserialize, CandidType)]
+struct Metadata {
+    description: Option<String>, // Updated to Option
+    date: Option<String>,        // Added date field
+    place: Option<String>,
+    tags: Option<Vec<String>>,
+    people: Option<Vec<String>>,
+    visibility: Option<Vec<Principal>>,
+}
+
+/* TODO: Consider size limitations for large image data. Since images can be large,
+     we may need to implement chunking or compression to ensure the serialized data
+     fits within the limits of stable memory.
+*/
 
+// Serialize and store the state before upgrading
+#[pre_upgrade]
+fn pre_upgrade() {
```

Comment:
I had to comment out pre_upgrade and post_upgrade to deploy

Reply:
Why? Which errors did you get?